reference/matrix/dense_kernels.cpp

```diff
@@ -296,6 +296,26 @@ void compute_dot(std::shared_ptr<const ReferenceExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_DENSE_COMPUTE_DOT_KERNEL);
 
 
+template <typename ValueType>
+void compute_dot_vendor(std::shared_ptr<const ReferenceExecutor> exec,
+                        const matrix::Dense<ValueType>* x,
+                        const matrix::Dense<ValueType>* y,
+                        matrix::Dense<ValueType>* result)
+{
+    for (size_type j = 0; j < x->get_size()[1]; ++j) {
```

Comment:
Haha, this is funny. Can we just optionally link to Netlib blas and call that routine here instead? If we don't find a system Blas, we could leave this unimplemented. For testing, we already have `compute_dot` to compare against.